src/components/is-scrolling-up.mixin.ts

```diff
@@ -0,0 +1,38 @@
+import { ScrollDirection, XOn } from '@empathyco/x-components';
+import Vue from 'vue';
+import Component from 'vue-class-component';
+@Component
+export default class IsScrollingUp extends Vue {
+  protected stopsScrollDown = true;
+  protected isScrollingUp = true;
+  protected scrollOffset = 200;
+
+  @XOn('UserChangedScrollDirection')
+  scrollDirectionChanged(direction: ScrollDirection): void {
+    this.stopsScrollDown = direction === 'UP';
+  }
+
+  @XOn(['UserScrolled'])
+  scrollLength(position: number): void {
```

Comment:
Convert to watcher and rename function

```suggestion
  @Watch('mainScrollPosition', { deep: true })
  updateHasScrolledPastThreshold(): void {
```